### PR TITLE
REST Polling Implementation

### LINKED ISSUE #6: Implement Rest Polling
At the moment the conversion rates are hardcoded in this file:
image

### Tasks to complete
1. Research free API services that provides these conversion rates.
2. Implement it in the APP - every 5 minutes, the app should fetch the conversion rates.

There will be a follow-up issue for another functionality - admins to configure how often the app will be refreshing the rates.
This is just FYI of what we would do in the future.

### PR DESCRIPTION
**Pull request by TheaAleksovDev**

### CODE REVIEW

**Comment by valentin-aleksandrov** [comment]
@TheaAleksovDev - I've changed the destination branch to `feature/budget-calculator`.
Once `feature/budget-calculator` gets merged, this PR will automatically switch its destination branch to main.
The reason we are doing this is because you have branch off `feature/budget-calculator` and in this PR there were changes from that PR, which is making the review process more difficult.

When we need changes from other PRs we do one of the following:
1. wait for that other PR to me merged
      - The risk here is that we are waiting and managers may not be happy - this is something, which we discuss we the manager and we decide based on priority.
2. branch off the PR 
     - With this approach you can start right off, but if the PR that we have branch off has some changes, then you would need to rebase the new branch against the new version the PR that we are depending on. It's not impossible, but it may take additional time.
     - In this case I did it, because I've already approved the `feature/budget-calculator` PR and I would not expect changes to be made there.

File: src/app/currency.pipe.ts
Comment (valentin-aleksandrov) [inline src/app/currency.pipe.ts]: **Optional**:
```suggestion
    rates: [Rate, Rate]
```

File: src/app/services/conversion-rates.service.ts
Comment (valentin-aleksandrov) [inline src/app/services/conversion-rates.service.ts]: **300000** should be a constant with a meaningful name.

File: src/app/services/conversion-rates.service.ts
Comment (valentin-aleksandrov) [inline src/app/services/conversion-rates.service.ts]:
```suggestion
  ratesInterval$ = new BehaviorSubject<number | null>(null);
```

**Comment by valentin-aleksandrov** [comment]
It seems that this API has a limit:
image

**Comment by TheaAleksovDev** [comment]
> It seems that this API has a limit: image

I found that this API has a header "X-RateLimit-Limit-Quota-Minute: 10"
In my latest commit, I solved the issue by checking if the input interval is greater than 10000ms (which is slightly greater than the exact limit of 6000ms)

**Review by valentin-aleksandrov — APPROVED** [review]

**Pull request merged**
